nova_vm/src/ecmascript/types/language/value.rs

```diff
@@ -290,9 +290,12 @@ impl From<bool> for Value {
     }
 }
 
-impl From<Option<Value>> for Value {
-    fn from(value: Option<Value>) -> Self {
-        value.unwrap_or(Value::Undefined)
+impl<T> From<Option<T>> for Value
+where
+    T: Into<Value>,
+{
+    fn from(value: Option<T>) -> Self {
+        value.map_or(Value::Undefined, |v| v.into())
```

Comment:
**thought:** I wonder if this is a good idea. There might be some places where we want to "default" into `Null` instead of `Undefined`.

That being said, we can just make a helper for it at that time.